src/test/java/org/assertj/core/internal/throwables/Throwables_assertHasRootCause_Test.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2018 the original author or authors.
+ */
+package org.assertj.core.internal.throwables;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.assertj.core.error.ShouldHaveCause.shouldHaveCause;
+import static org.assertj.core.error.ShouldHaveNoCause.shouldHaveNoCause;
+import static org.assertj.core.error.ShouldHaveRootCause.shouldHaveRootCause;
+import static org.assertj.core.test.TestData.someInfo;
+import static org.assertj.core.test.TestFailures.failBecauseExpectedAssertionErrorWasNotThrown;
+import static org.assertj.core.util.AssertionsUtil.expectAssertionError;
+import static org.assertj.core.util.FailureMessages.actualIsNull;
+import static org.mockito.Mockito.verify;
+
+import org.assertj.core.api.AssertionInfo;
+import org.assertj.core.internal.ThrowablesBaseTest;
+import org.junit.jupiter.api.Test;
+
+public class Throwables_assertHasRootCause_Test extends ThrowablesBaseTest {
+
+  @Test
```

Comment:
I think the `should_pass...` tests can be combined as a parameterized test for the tests where WHEN and THEN sections are the same.

JUnit 5 comes with great support for parameterized tests, I'm starting to use that more and more.
I suggest following this test example: https://github.com/joel-costigliola/assertj-core/blob/recursive-api/src/test/java/org/assertj/core/internal/objects/Objects_assertIsEqualToUsingRecursiveComparison_Test.java#L219.

Since we are combining tests, we will be loosing their names that reflect the assertion contract, we can keep that contract documentation by adding a test description as one of the parameter (as shown in the link above).